packages/kbn-unified-data-table/src/components/data_table_columns.tsx

```diff
@@ -267,6 +268,21 @@ export function getEuiGridColumns({
   );
 }
 
+export function hasSourceTimeFieldValue(
+  columns: string[],
+  dataView: DataView,
+  rows: DataTableRecord[],
+  showTimeCol: boolean,
+  isPlainRecord: boolean
+) {
+  const timeFieldName = dataView.timeFieldName;
+  if (!isPlainRecord || !columns.includes('_source') || !timeFieldName) {
+    return showTimeCol;
+  }
+  // flattened is the _source content in this case
+  return rows.some((row) => timeFieldName in row.flattened);
```

Comment:
Iterating through all results can be a long operation. Can we avoid it?

Reply:
For example, we could search in the returned ESQL fields. UnifiedDataTable accesses them via `columnTypes`. So in your case it could be:
```
return Boolean(columnTypes[timeFieldName])
```